packages/forklift-console-plugin/src/modules/Providers/views/details/tabs/VirtualMachines/components/PowerStateCellRenderer.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { TableCell } from 'src/modules/Providers/utils';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { VSphereVM } from '@kubev2v/types';
+import { Tooltip } from '@patternfly/react-core';
+import { OffIcon, PowerOffIcon, UnknownIcon } from '@patternfly/react-icons';
+
+import { VMCellProps } from './VMCellProps';
+
+export const PowerStateCellRenderer: React.FC<VMCellProps> = ({ data }) => {
+  const powerState = (data?.vm as VSphereVM)?.powerState;
+  const { t } = useForkliftTranslation();
+  const [icon, tooltipText] =
+    powerState === 'poweredOn'
+      ? [<PowerOffIcon color="green" key="on" />, t('Powered on')]
+      : powerState === 'poweredOff'
+      ? [<OffIcon color="red" key="off" />, t('Powered off')]
+      : [<UnknownIcon key="unknown" />, t('Unknown power state')];
```

Comment:
```suggestion
const powerStateDetails = {
  poweredOn: {
    icon: <PowerOffIcon color="green" key="on" />,
    tooltip: t('Powered on')
  },
  poweredOff: {
    icon: <OffIcon color="red" key="off" />,
    tooltip: t('Powered off')
  },
  default: {
    icon: <UnknownIcon key="unknown" />,
    tooltip: t('Unknown power state')
  }
};

const { icon, tooltip } = powerStateDetails[powerState] || powerStateDetails.default;
```

`powerStateDetails` can be defined elsewhere, or you can even make a helper method `getPowerStateDetails(powerState: string) `

Reply:
the `getPowerStateDetails` has an aditional advantage, it can make this method work for more then one type, as it can return other icons and texts for other types of providers depending on there entire object and not just the powerState field

Reply:
The code has been extracted to generic method (borrowed from [the legacy](https://github.com/kubev2v/forklift-console-plugin/blob/8c17951019a08a33f55b7c820e694551dd99c6dc/packages/legacy/src/common/components/VMNameWithPowerState.tsx#L24)).  Let me know if we need to split it more.